connectors/agentic-ai/src/test/java/io/camunda/connector/agenticai/a2a/client/inbound/webhook/A2aClientWebhookExecutableTest.java

```diff
@@ -0,0 +1,397 @@
+    var result = webhookExecutable.triggerWebhook(payload);
+
+    assertNotNull(result);
+    assertThat(result.request().body()).isInstanceOf(A2aTask.class);
+  }
+
+  @Test
+  void triggerWebhook_HmacEnabled_InvalidSignature_ThrowsException() {
+    String secret = "mySecret123";
+
+    InboundConnectorContext ctx =
+        InboundConnectorContextBuilder.create()
+            .properties(
+                Map.of(
+                    "inbound",
+                    Map.of(
+                        "context",
+                        "a2aWebhookContext",
+                        "clientResponse",
+                        "=task",
+                        "auth",
+                        Map.of("type", "NONE"),
+                        "shouldValidateHmac",
+                        enabled.name(),
+                        "hmacSecret",
+                        secret,
+                        "hmacHeader",
+                        "X-Hub-Signature",
+                        "hmacAlgorithm",
+                        HMACAlgoCustomerChoice.sha_256.name())))
+            .build();
+
+    WebhookProcessingPayload payload = mock(WebhookProcessingPayload.class);
+    when(payload.method()).thenReturn(HttpMethods.post.name());
+    when(payload.headers())
+        .thenReturn(
+            Map.of(
+                HttpHeaders.CONTENT_TYPE,
+                MediaType.JSON_UTF_8.toString(),
+                "X-Hub-Signature",
+                "invalid-signature"));
+    when(payload.rawBody()).thenReturn(TASK_JSON.getBytes(StandardCharsets.UTF_8));
+
+    webhookExecutable.activate(ctx);
+
+    var exception = catchException(() -> webhookExecutable.triggerWebhook(payload));
+
+    assertThat(exception).isInstanceOf(WebhookSecurityException.class);
+    assertThat(exception).hasMessageContaining("HMAC signature check didn't pass");
+  }
+
+  @Test
+  void triggerWebhook_HmacDisabled_NoSignatureValidation() {
+    InboundConnectorContext ctx =
+        InboundConnectorContextBuilder.create()
+            .properties(
+                Map.of(
+                    "inbound",
+                    Map.of(
+                        "context",
+                        "a2aWebhookContext",
+                        "clientResponse",
+                        "=task",
+                        "auth",
+                        Map.of("type", "NONE"),
+                        "shouldValidateHmac",
+                        disabled.name())))
+            .build();
+
+    WebhookProcessingPayload payload = mock(WebhookProcessingPayload.class);
+    when(payload.method()).thenReturn(HttpMethods.post.name());
+    when(payload.headers())
+        .thenReturn(Map.of(HttpHeaders.CONTENT_TYPE, MediaType.JSON_UTF_8.toString()));
+    when(payload.rawBody()).thenReturn(TASK_JSON.getBytes(StandardCharsets.UTF_8));
+
+    webhookExecutable.activate(ctx);
+    var result = webhookExecutable.triggerWebhook(payload);
+
+    assertNotNull(result);
```

Comment:
🔧 Nit: I think we usually use mostly AssertJ assertions, could we stick to AssertJ for these assertions as well (`assertThat(..).isNotNull()`)?